shopfloor/actions/stock.py

```diff
@@ -10,6 +13,96 @@ class StockAction(Component):
     _inherit = "shopfloor.process.action"
     _usage = "stock"
 
+    def _create_return_move__get_max_qty(self, origin_move):
+        quantity = origin_move.product_qty
+        for move in origin_move.move_dest_ids:
+            if (
+                move.origin_returned_move_id
+                and move.origin_returned_move_id != origin_move
+            ):
+                continue
+            if move.state in ("partially_available", "assigned"):
+                quantity -= sum(move.move_line_ids.mapped("product_qty"))
+            elif move.state in ("done"):
+                quantity -= move.product_qty
+        return float_round(
+            quantity, precision_rounding=origin_move.product_id.uom_id.rounding
+        )
+
+    def _create_return_move__get_vals(self, return_picking, origin_move):
+        product = origin_move.product_id
+        return_type = return_picking.picking_type_id
+        return {
+            "product_id": product.id,
+            "product_uom": product.uom_id.id,
+            "picking_id": return_picking.id,
+            "state": "draft",
+            "date": fields.Datetime.now(),
+            "location_id": return_picking.location_id.id,
+            "location_dest_id": return_picking.location_dest_id.id,
```

Comment:
I added a check (not yet pushed) at return picking creation.
This will prevent to create a picking with empty locations (which are used in the rest of the flow).